### PR TITLE
Group sphinx definitions

### PR DESCRIPTION
**Pull request by nineteendo**
Issue: https://github.com/python/cpython/issues/118030

----
📚 Documentation preview 📚: https://cpython-devguide--1322.org.readthedocs.build/

### CODE REVIEW

**Comment by nineteendo** [comment]
@hugovk, should we add it here?

**Comment by hugovk** [comment]
For reference:

- https://github.com/python/cpython/issues/118030
- https://discuss.python.org/t/grouping-sphinx-object-definition-directives/52745

File: documentation/markup.rst [hunk 1]
```diff
@@ -246,14 +246,16 @@ directives.)  Looking at this example,
 
 ::
 
-   .. function:: foo(x)
-                 foo(y, z)
+   .. function:: remove(path, *, dir_fd=None)
+                 unlink(path, *, dir_fd=None)
```
Comment (ezio-melotti) [inline documentation/markup.rst:250]: It seems there are at least two use cases for this:
1. showing different names/aliases for the same function (like you did)
2. showing multiple signatures for the same function (the original example)

I think the second one is much more common and might be a better example.  Just looking at the [built-in functions](https://docs.python.org/3/library/functions.html) there are several -- such as [`int`](https://docs.python.org/3/library/functions.html#int) or [`max`](https://docs.python.org/3/library/functions.html#max) -- that could be used if we wanted a real-world example.

File: documentation/markup.rst [hunk 2]
```diff
@@ -246,14 +246,16 @@ directives.)  Looking at this example,
...
       :bar: no
```
Comment (ezio-melotti) [inline documentation/markup.rst:251]: Leaving this after updating the example looks a bit out of place though.  Are there any real-world examples that have options?
Otherwise why not just leave the original example unchanged?

File: documentation/markup.rst [hunk 3]
```diff
@@ -246,14 +246,16 @@ directives.)  Looking at this example,
...
 
-      Return a line of text input from the user.
+      Remove (delete) the file *path*.
 
 ``function`` is the directive name.  It is given two arguments here, the
 remainder of the first line and the second line, as well as one option ``bar``
+By providing multiple arguments, the definitions will be grouped instead of
+separated by a empty line.
```
Comment (ezio-melotti) [inline documentation/markup.rst:258]: Here you technically provided multiple *names* with the same *arguments*.  The original example had the same *name* (twice) and multiple different *arguments*.
I'm not sure if there is a term that applies to both the *name* and the *argument* -- maybe *lines* is ok:
```suggestion
By providing multiple lines, the definitions will be grouped instead of
separated by a empty line.
```

File: documentation/markup.rst (hunk 2 above)
Comment (nineteendo) [reply documentation/markup.rst]: Yeah, let's keep the orignal example.

File: documentation/markup.rst (hunk 1 above)
Comment (nineteendo) [reply documentation/markup.rst]: I explained the second use in text only.

**Comment by nineteendo** [comment]
Thanks Ezio.

**Comment by nineteendo** [comment]
Can this be merged? It should happen before group other definitions in the docs.

File: documentation/markup.rst [hunk 4]
```diff
@@ -429,7 +432,7 @@ following example shows all of the features of this directive type::
                   ham(eggs)
        :noindex:
 
-       Spam or ham the foo.
+       Spam or ham the eggs.
```
Comment (erlend-aasland) [inline documentation/markup.rst:435]: Please revert unrelated changes.

```suggestion
       Spam or ham the foo.
```

Comment (nineteendo) [reply documentation/markup.rst:435]: Alright, let's leave it then, it's too minor for another pull request.

**Comment by nineteendo** [comment]
Do we merge this or not?

**Comment by nineteendo** [comment]
Let's close this, it probably doesn't solve the initial issue.
